[CI] GenerativeRandomMappingIT test failing

**Build Scans:**
- [elasticsearch-intake #38009 / part3](https://gradle-enterprise.elastic.co/s/iovyqwelbbfya)
- [elasticsearch-intake #38001 / part3](https://gradle-enterprise.elastic.co/s/rkfka7ncp3y5u)
- [elasticsearch-intake #37994 / part3](https://gradle-enterprise.elastic.co/s/3z75keihde5oe)
- [elasticsearch-pull-request #138876 / part-3](https://gradle-enterprise.elastic.co/s/lpgba66tn4haq)
- [elasticsearch-pull-request #138872 / part-3](https://gradle-enterprise.elastic.co/s/ojpiv7keezsbg)
- [elasticsearch-pull-request #138876 / part-3](https://gradle-enterprise.elastic.co/s/ydbqhtjnnfh7i)
- [elasticsearch-pull-request #138872 / part-3](https://gradle-enterprise.elastic.co/s/nq2blau7wzgju)
- [elasticsearch-intake #37805 / part3](https://gradle-enterprise.elastic.co/s/4umtvs4m4kbx4)
- [elasticsearch-pull-request #138261 / part-3](https://gradle-enterprise.elastic.co/s/5zhvamb5nfzro)
- [elasticsearch-pull-request #138261 / part-3](https://gradle-enterprise.elastic.co/s/xrlrksu4xnedg)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.GenerativeRandomMappingIT.test" -Dtests.seed=51F625780048765B -Dtests.locale=en-ZM -Dtests.timezone=America/Detroit -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.GenerativeRandomMappingIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: query: from gen_rm_0,ge*
exception: method [POST], host [http://[::1]:42439], URI [/_query?pretty=true&error_trace=true], status line [HTTP/1.1 500 Internal Server Error]
Warnings: [No limit defined, adding default limit of [1000]]
---
error:
  root_cause:
  - type: "illegal_state_exception"
    reason: "Cannot load blocks without doc values"
    stack_trace: "org.elasticsearch.ElasticsearchException$1: Cannot load blocks without\
      \ doc values\n\tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:810)\n\
      \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:724)\n\
      \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.build(RestResponse.java:202)\n\
      \tat org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.<init>(RestResponse.jav
[truncated]
```

**Issue Reasons:**
- [main] 1 consecutive failures in test test
- [main] 1 consecutive failures in step part3
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 36 failures in test test (14.9% fail rate in 241 executions)
- [main] 8 failures in step part3 (11.3% fail rate in 71 executions)
- [main] 26 failures in step part-3 (19.7% fail rate in 132 executions)
- [main] 2 failures in step part-3-arm (50.0% fail rate in 4 executions)
- [main] 8 failures in pipeline elasticsearch-intake (11.3% fail rate in 71 executions)
- [main] 13 failures in pipeline elasticsearch-pull-request (10.8% fail rate in 120 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -325,9 +325,6 @@ tests:
 - class: org.elasticsearch.index.reindex.ReindexRelocationOnShutdownIT
   method: testReindexFailsWhenPitRelocationFails
   issue: https://github.com/elastic/elasticsearch/issues/146650
-- class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeRandomMappingIT
-  method: test
-  issue: https://github.com/elastic/elasticsearch/issues/146553
 - class: org.elasticsearch.index.reindex.ReindexRelocationOnShutdownIT
   method: testReindexTaskRelocatesOnNodeShutdown
   issue: https://github.com/elastic/elasticsearch/issues/146696

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/generator/RandomMappingGenerator.java
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/generator/RandomMappingGenerator.java
@@ -281,8 +281,7 @@ public final class RandomMappingGenerator {
                 boolean indexed = Boolean.FALSE.equals(s.get("index")) == false;
                 Object indexOpts = s.get("index_options");
                 // omitted index_options defaults to "positions" (see TextParams.textIndexOptions)
-                boolean positionsEnabled = indexed
-                    && (indexOpts == null || "positions".equals(indexOpts) || "offsets".equals(indexOpts));
+                boolean positionsEnabled = indexed && (indexOpts == null || "positions".equals(indexOpts) || "offsets".equals(indexOpts));
                 if (positionsEnabled) {
                     optionalSet(s, 1, "index_phrases", true);
                     optionalSet(s, 1, "position_increment_gap", randomFrom(100, 0, 50, 200));

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/generator/RandomMappingGenerator.java
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/generator/RandomMappingGenerator.java
@@ -278,12 +278,21 @@ public final class RandomMappingGenerator {
                 optionalSet(s, 1, "index", randomBoolean());
                 optionalSet(s, 1, "index_options", randomFrom("docs", "freqs", "positions", "offsets"));
                 optionalSet(s, 1, "norms", randomBoolean());
-                optionalSet(s, 1, "index_phrases", true);
-                optionalSet(s, 1, "position_increment_gap", randomFrom(100, 0, 50, 200));
+                boolean indexed = Boolean.FALSE.equals(s.get("index")) == false;
+                Object indexOpts = s.get("index_options");
+                // omitted index_options defaults to "positions" (see TextParams.textIndexOptions)
+                boolean positionsEnabled = indexed
+                    && (indexOpts == null || "positions".equals(indexOpts) || "offsets".equals(indexOpts));
+                if (positionsEnabled) {
+                    optionalSet(s, 1, "index_phrases", true);
+                    optionalSet(s, 1, "position_increment_gap", randomFrom(100, 0, 50, 200));
+                }
                 optionalSet(s, 1, "similarity", randomFrom("BM25", "boolean"));
                 optionalSet(s, 1, "term_vector", randomFrom("no", "yes", "with_positions", "with_offsets", "with_positions_offsets"));
                 optionalSet(s, 1, "eager_global_ordinals", true);
-                optionalSet(s, 1, "fielddata", true);
+                if (indexed) {
+                    optionalSet(s, 1, "fielddata", true);
+                }
             }
             case "match_only_text" -> {
             }

